high_card.rb

```diff
@@ -3,3 +3,47 @@
 suits = [ "hearts", "spades", "clubs", "diamonds" ]
 
 players = []
+deck = []
+n = players.length
+ranks.length.times do |rank_index|
+  suits.length.times do |suit_index|
+    deck.push [ ranks[rank_index], suits[suit_index] ]
+
+  end
+end
+p deck.shuffle!
```

Comment:
in general I would advise against mutation, in favor of reassignment, i.e. `deck = deck.shuffle`